src/Microsoft.Health.SqlServer/Features/Client/SqlConnectionWrapper.cs

```diff
@@ -17,55 +16,74 @@ public class SqlConnectionWrapper : IDisposable
         private readonly bool _enlistInTransactionIfPresent;
         private readonly SqlTransactionHandler _sqlTransactionHandler;
         private readonly SqlCommandWrapperFactory _sqlCommandWrapperFactory;
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
+
+        private SqlConnection _sqlConnection;
+        private SqlTransaction _sqlTransaction;
 
         public SqlConnectionWrapper(
-            SqlServerDataStoreConfiguration configuration,
             SqlTransactionHandler sqlTransactionHandler,
             SqlCommandWrapperFactory sqlCommandWrapperFactory,
+            ISqlConnectionFactory sqlConnectionFactory,
             bool enlistInTransactionIfPresent)
         {
-            EnsureArg.IsNotNull(configuration, nameof(configuration));
             EnsureArg.IsNotNull(sqlTransactionHandler, nameof(sqlTransactionHandler));
             EnsureArg.IsNotNull(sqlCommandWrapperFactory, nameof(sqlCommandWrapperFactory));
+            EnsureArg.IsNotNull(sqlConnectionFactory, nameof(sqlConnectionFactory));
 
             _sqlTransactionHandler = sqlTransactionHandler;
             _enlistInTransactionIfPresent = enlistInTransactionIfPresent;
             _sqlCommandWrapperFactory = sqlCommandWrapperFactory;
+            _sqlConnectionFactory = sqlConnectionFactory;
+
+            InitializeAsync();
+        }
+
+        public SqlConnection SqlConnection
+        {
+            get { return _sqlConnection; }
+        }
+
+        public SqlTransaction SqlTransaction
+        {
+            get
+            {
+                return _sqlTransaction;
+            }
+        }
 
-            if (_enlistInTransactionIfPresent && sqlTransactionHandler.SqlTransactionScope?.SqlConnection != null)
+        public async void InitializeAsync()
+        {
+            if (_enlistInTransactionIfPresent && _sqlTransactionHandler.SqlTransactionScope?.SqlConnection != null)
             {
-                SqlConnection = sqlTransactionHandler.SqlTransactionScope.SqlConnection;
+                _sqlConnection = _sqlTransactionHandler.SqlTransactionScope.SqlConnection;
             }
             else
             {
-                SqlConnection = new SqlConnection(configuration.ConnectionString);
+                _sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync();
             }
 
-            if (_enlistInTransactionIfPresent && sqlTransactionHandler.SqlTransactionScope != null && sqlTransactionHandler.SqlTransactionScope.SqlConnection == null)
+            if (_enlistInTransactionIfPresent && _sqlTransactionHandler.SqlTransactionScope != null && _sqlTransactionHandler.SqlTransactionScope.SqlConnection == null)
             {
-                sqlTransactionHandler.SqlTransactionScope.SqlConnection = SqlConnection;
+                _sqlTransactionHandler.SqlTransactionScope.SqlConnection = SqlConnection;
             }
 
             if (SqlConnection.State != ConnectionState.Open)
             {
```

Comment:
this can be OpenAsync now

Reply:
True. I updated it before and with refactoring, I mistakenly undid my changes. Will fix.